cobalt/persistent_storage/persistent_settings.cc

```diff
@@ -96,11 +76,19 @@ PersistentSettings::~PersistentSettings() {
   // the destruction observer to be notified.
   writeable_pref_store_initialized_.Wait();
   destruction_observer_added_.Wait();
+  // Wait for all previously posted tasks to finish.
+  thread_.message_loop()->task_runner()->WaitForFence();
   thread_.Stop();
```

Comment:
reset results in stop, since it's a member and this is the destructor, it will also reset, but for consistency with other code I can change this into a reset.

Reply:
Oh, this is not a unique_ptr, it's a regular member, so can't be reset.